src/annotator/test/highlighter-test.js

```diff
@@ -122,6 +122,23 @@ describe('annotator/highlighter', () => {
       });
     });
 
+    it('generates correct highlights when the start text node is split', () => {
+      const el = document.createElement('span');
+      el.append('foo bar baz');
+
+      // nb. It is important for this test case that the start is in the middle
```

Comment:
I think it would be helpful to have a broader comment here that illustrates the DOM and range being acted upon.